### PR TITLE
feat(CE): Added Mailchimp Destination Connector

### PR DESCRIPTION
**Pull request by xyfer17**
## Description

- Add MailChimp Destination Connector
- Add Unit Test for MailChimp Connector

## Related Issue

Relates to issue #391 Mailchimp: Destination Connector

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] New Connector (Destination or Source Connector)
- [ ] Breaking change (fix or feature that would impact existing functionality)
- [ ] Styling change
- [ ] Refactoring
- [ ] Documentation update
- [ ] Chore

## Checklist:

- [x] Added unit tests for the changes made (if required)
- [x] Ensure a branch name is prefixed with `feature`, `bugfix`, `hotfix`, `release`, `style` or `chore` followed by `/` and branch name e.g `feature/add-salesforce-connector`
- [x] Have you made sure the commit messages meets the guidelines?
- [ ] Added relevant screenshots for the changes
- [x] Have you tested the changes on local/staging?
- [x] Added the new connector in rollout.rb
- [x] Have you updated the version number of the gem?
- [ ] Have you ensured that your changes for new connector are documented in the [docs repo](https://github.com/Multiwoven/docs) or relevant documentation files?
- [x] Have you updated the run time dependency in multiwoven-integrations.gemspec if new gems are added
- [x] Have you made sure the code you have written follows the best practises to the best of your knowledge?

### CODE REVIEW

**Comment by xyfer17** [comment]
@afthabvp have raised draft pr for mailchimp connector, please take look.

File: integrations/lib/multiwoven/integrations/destination/mailchimp/config/spec.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        "multiwoven_secret": true,
+        "title": "API Key",
+        "order": 0
+      },
+      "list_id": {
+        "type": "string",
+        "title": "List Id",
+        "order": 1
+      },
+      "email_template_id": {
```
Comment (afthabvp) [inline integrations/lib/multiwoven/integrations/destination/mailchimp/config/spec.json:21]: @xyfer17 what is email_template_id?
is it mandatory or specific to any stream?

Comment (xyfer17) [reply integrations/lib/multiwoven/integrations/destination/mailchimp/config/spec.json:21]: `email_template_id ` is the template id for the email campaigns, if user have email template already then it is used, as it's not mandatory.

File: integrations/lib/multiwoven/integrations/destination/mailchimp/client.rb [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+          rescue StandardError => e
+            handle_exception(e, {
+                               context: "MAILCHIMP:DISCOVER:EXCEPTION",
+                               type: "error"
+                             })
+          end
+
+          def write(sync_config, records, action = "create")
+            @sync_config = sync_config
+            @action = sync_config.stream.action || action
```
Comment (TivonB-AI2) [inline integrations/lib/multiwoven/integrations/destination/mailchimp/client.rb:35]: @xyfer17 Minor nitpick, is `@action` used for anything?

File: integrations/lib/multiwoven/integrations/destination/mailchimp/config/spec.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
+{
+  "documentation_url": "",
```
Comment (TivonB-AI2) [inline integrations/lib/multiwoven/integrations/destination/mailchimp/config/spec.json:2]: @xyfer17 Documentation_url is missing

File: integrations/spec/multiwoven/integrations/destination/mailchimp/client_spec.rb [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+    let(:list_member_url) { "https://#{server}.api.mailchimp.com/3.0/lists/#{list_id}/members" }
+
+    context "when the write operation is successful" do
+      before do
+        # Mock the request to the Mailchimp API for adding/updating a list member
+        stub_request(:put, "#{list_member_url}/#{Digest::MD5.hexdigest(records.first[:email].downcase)}")
+          .to_return(status: 200, body: "", headers: {})
+      end
+
+      it "increments the success count" do
```
Comment (TivonB-AI2) [inline integrations/spec/multiwoven/integrations/destination/mailchimp/client_spec.rb:120]: @xyfer17 This failed the Rspec check.

File: integrations/spec/multiwoven/integrations/destination/mailchimp/client_spec.rb [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+  private
+
+  def sync_config
+    Multiwoven::Integrations::Protocol::SyncConfig.from_json(
+      sync_config_json.to_json
+    )
+  end
+
+  describe "#discover" do
+    it "returns a catalog" do
```
Comment (TivonB-AI2) [inline integrations/spec/multiwoven/integrations/destination/mailchimp/client_spec.rb:169]: @xyfer17 This failed the Rspec check.

File: integrations/lib/multiwoven/integrations/rollout.rb [hunk 6]
```diff
@@ -41,6 +41,7 @@ module Integrations
       Oracle
       MicrosoftExcel
       MicrosoftSql
+      Mailchimp
```
Comment (TivonB-AI2) [inline integrations/lib/multiwoven/integrations/rollout.rb:44]: @xyfer17 I believe the Version should increase here. @afthabvp Second opinion here

File: integrations/lib/multiwoven/integrations/destination/mailchimp/client.rb (hunk 2 above)
Comment (xyfer17) [reply integrations/lib/multiwoven/integrations/destination/mailchimp/client.rb]: fixed in latest commit

File: integrations/lib/multiwoven/integrations/destination/mailchimp/config/spec.json (hunk 3 above)
Comment (xyfer17) [reply integrations/lib/multiwoven/integrations/destination/mailchimp/config/spec.json]: fixed in latest commit

File: integrations/spec/multiwoven/integrations/destination/mailchimp/client_spec.rb (hunk 5 above)
Comment (xyfer17) [reply integrations/spec/multiwoven/integrations/destination/mailchimp/client_spec.rb:169]: fixed in latest commit

File: integrations/spec/multiwoven/integrations/destination/mailchimp/client_spec.rb [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+    end
+
+    context "when the write operation fails" do
+      before do
+        # Mock the request to simulate a failure when trying to add/update a list member
+        stub_request(:put, "#{list_member_url}/#{Digest::MD5.hexdigest(records.first[:email].downcase)}")
+          .to_return(status: 400, body: '{"detail": "Invalid Request"}', headers: {})
+      end
+
+      it "increments the failure count" do
```
[the remark this answers is not in the document]
Comment (xyfer17) [reply integrations/spec/multiwoven/integrations/destination/mailchimp/client_spec.rb:139]: fixed in latest commit

File: integrations/spec/multiwoven/integrations/destination/mailchimp/client_spec.rb (hunk 4 above)
Comment (xyfer17) [reply integrations/spec/multiwoven/integrations/destination/mailchimp/client_spec.rb:119]: fixed in latest commit

File: integrations/lib/multiwoven/integrations/rollout.rb (hunk 6 above)
Comment (xyfer17) [reply integrations/lib/multiwoven/integrations/rollout.rb:44]: @afthabvp can you help me out for the above comment ?

Comment (afthabvp) [reply integrations/lib/multiwoven/integrations/rollout.rb:44]: @xyfer17 
https://github.com/Multiwoven/multiwoven/blob/93e8767684236e1eea9cb55b6e9719de14afd671/integrations/lib/multiwoven/integrations/rollout.rb#L5
for each new connector need to update version 0.12.0 -> 0.13.0
once this version updated in gem we need to update the server gem

Comment (xyfer17) [reply integrations/lib/multiwoven/integrations/rollout.rb:44]: @afthabvp have updated the version in commit  https://github.com/Multiwoven/multiwoven/pull/437/commits/af0c76df7fc88f170578433d2d9ab507edccbc83

**Review by afthabvp — APPROVED** [review]

**Pull request merged**